ducktape/tests/test.py

```diff
@@ -384,8 +385,11 @@ def expected_cluster_spec(self):
 
         :return:            A ClusterSpec object.
         """
+        cluster_spec = self.cluster_use_metadata.get(CLUSTER_SPEC_KEYWORD)
         cluster_size = self.cluster_use_metadata.get(CLUSTER_SIZE_KEYWORD)
-        if cluster_size is not None:
+        if cluster_spec is not None:
```

Comment:
if cluster_spec provided, return it.
else if nums_nodes provided, use that to create ClusterSpec accordingly.
else if test context not successfully initialized, expect 0 node because test is not ready.
else test is ready, but no valid hint for resource available, set expected_nums_nodes to number of nodes we have.

Reply:
@ewencp

Reply:
Done!